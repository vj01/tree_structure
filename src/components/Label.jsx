import React from "react";
import "../styles/tree.css";

function Label(props) {
  return (<div className="label-div">
    <span data-testid="titleTag">{props.title}</span>
  </div>);
}
export default Label;
