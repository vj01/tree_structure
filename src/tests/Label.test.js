import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Label from '../components/Label';

afterEach(cleanup);

test('Label component modified', () => {
  const { asFragment } = render(
    <Label />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Insert text at label", () => {
  const { getByTestId } = render(<Label title="Node text!" />);

  expect(getByTestId("titleTag")).toHaveTextContent("Node text!");
});
