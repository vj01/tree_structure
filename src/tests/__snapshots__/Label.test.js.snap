// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component modified 1`] = `
<DocumentFragment>
  <div
    class="label-div"
  >
    <span
      data-testid="titleTag"
    />
  </div>
</DocumentFragment>
`;
